frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseAction.java

```diff
@@ -0,0 +1,287 @@
+package com.virtuslab.gitmachete.frontend.actions.traverse;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.actionSystem.ActionManager;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.ui.MessageDialogBuilder;
+import com.intellij.openapi.ui.Messages;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.i18nformatter.qual.I18nConversionCategory;
+import org.checkerframework.checker.i18nformatter.qual.I18nFormat;
+import org.checkerframework.checker.tainting.qual.Untainted;
+
+import com.virtuslab.gitmachete.backend.api.IManagedBranchSnapshot;
+import com.virtuslab.gitmachete.backend.api.IRemoteTrackingBranchReference;
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus;
+import com.virtuslab.gitmachete.frontend.actions.base.BaseGitMacheteRepositoryReadyAction;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.DivergedFromParentDialog;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.DivergedFromRemoteDialog;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.PullApprovalDialog;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.PushApprovalDialog;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.TraverseApprovalDialog;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
+import com.virtuslab.gitmachete.frontend.actions.navigation.CheckoutNextAction;
+import com.virtuslab.gitmachete.frontend.actions.navigation.CheckoutPreviousAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.OverrideForkPointOfCurrentAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.PullCurrentAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.PushCurrentAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.RefreshStatusAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.ResetCurrentToRemoteAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.SlideOutCurrentAction;
+import com.virtuslab.gitmachete.frontend.actions.toolbar.SyncCurrentToParentByRebaseAction;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+import com.virtuslab.qual.gitmachete.backend.api.ConfirmedNonRoot;
+
+@ExtensionMethod({GitVfsUtils.class, GitMacheteBundle.class})
+@CustomLog
+public class TraverseAction extends BaseGitMacheteRepositoryReadyAction implements IExpectsKeySelectedBranchName {
```

Comment:
do we need `IExpectsKeySelectedBranchName` here? I do not see any reason for that but maybe I miss something